x/application/keeper/msg_server_undelegate_from_gateway.go

```diff
@@ -49,6 +51,34 @@ func (k msgServer) UndelegateFromGateway(ctx context.Context, msg *types.MsgUnde
 	// Remove the gateway from the application's delegatee gateway public keys
 	foundApp.DelegateeGatewayAddresses = append(foundApp.DelegateeGatewayAddresses[:foundIdx], foundApp.DelegateeGatewayAddresses[foundIdx+1:]...)
 
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	currentBlock := sdkCtx.BlockHeight()
+	sessionEndHeight := uint64(sessionkeeper.GetSessionEndBlockHeight(currentBlock))
+
+	// TODO_INVESTIGATE: When there is no undelegation, the undelegations map is nil
+	// even though it was declared with nullable=false in the application proto.
+	if foundApp.Undelegations == nil {
+		foundApp.Undelegations = make(map[uint64]types.UndelegationFromAppToGatewayEvent)
+	}
+
+	// Create a session undelegation entry for the given session end height if it doesn't exist.
+	undelegationsAtBlock, ok := foundApp.Undelegations[sessionEndHeight]
+	if !ok {
+		undelegationsAtBlock = types.UndelegationFromAppToGatewayEvent{
+			UndelegatedGateways: []string{},
+		}
+	}
+
+	// Add the gateway to the undelegated gateways list if it's not already there.
+	if !slices.Contains(undelegationsAtBlock.UndelegatedGateways, msg.GatewayAddress) {
+		undelegationsAtBlock.UndelegatedGateways = append(
+			undelegationsAtBlock.UndelegatedGateways,
+			msg.GatewayAddress,
+		)
+
+		foundApp.Undelegations[sessionEndHeight] = undelegationsAtBlock
```

Comment:
I don't think this line is necessary since slices are pointer. Can you remove and confirm?

Reply:
Will try this out :+1:

Reply:
Seems that it's still needed